From: Ben Hutchings <ben@decadent.org.uk>
Subject: grsecurity: Kconfig integration
Origin: https://grsecurity.net/test/grsecurity-3.1-4.1.3-201507261932.patch

Kconfig integration changes extracted from Grsecurity, with help
strings changed to make it clear that is not the real thing.

Subsequent patches will add to the empty menu.

---
--- /dev/null
+++ b/grsecurity/Kconfig
@@ -0,0 +1,3 @@
+#
+# grecurity configuration
+#
--- a/security/Kconfig
+++ b/security/Kconfig
@@ -4,6 +4,28 @@
 
 menu "Security options"
 
+menu "Hardening features (from Grsecurity)"
+
+config GRKERNSEC
+	bool "Hardening features (from Grsecurity)"
+	select DEBUG_KERNEL
+	select DEBUG_LIST
+	help
+	  If you say Y here, you will be able to configure many features
+	  that will enhance the security of your system.  It is highly
+	  recommended that you say Y here and read through the help
+	  for each option so that you fully understand the features and
+	  can evaluate their usefulness for your machine.
+
+menu "Customize Configuration"
+depends on GRKERNSEC
+
+source grsecurity/Kconfig
+
+endmenu
+
+endmenu
+
 source security/keys/Kconfig
 
 config SECURITY_DMESG_RESTRICT
